core/prisma/migrations/20241205152006_add_title_field/migration.sql

```diff
@@ -0,0 +1,75 @@
+-- AlterTable
+ALTER TABLE "pubs" ADD COLUMN     "title" TEXT;
+
+-- defined in core/prisma/migrations/20241126113759_add_pub_values_updated_at_trigger/migration.sql
+-- we are replacing it with a more complex function that also updates the title
+DROP TRIGGER IF EXISTS trigger_pub_values_update_pub ON "pub_values";
+DROP FUNCTION IF EXISTS update_pub_updated_at();
+
+CREATE OR REPLACE FUNCTION update_pub_for_value_changes()
+RETURNS TRIGGER AS $$
+BEGIN
+    CREATE TEMP TABLE IF NOT EXISTS tmp_affected_pubs (
+        "pubId" TEXT PRIMARY KEY,
+        "value" TEXT
+    ) ON COMMIT DROP;
+
+    WITH tf AS (
+        SELECT DISTINCT ON (inserted_deleted_rows."pubId") inserted_deleted_rows."pubId", inserted_deleted_rows."value" FROM inserted_deleted_rows 
+            JOIN "pubs" p ON inserted_deleted_rows."pubId" = p."id"
+            JOIN "_PubFieldToPubType" pft ON pft."A" = inserted_deleted_rows."fieldId" AND pft."B" = p."pubTypeId" AND pft."isTitle" = true
+    ) 
+    INSERT INTO tmp_affected_pubs ("pubId", "value")
+    -- the #>> '{}' transforms the json into text without the quotes
+    SELECT DISTINCT "pubId", ("value" #>> '{}') FROM tf
+    ON CONFLICT DO NOTHING;
+
+    UPDATE "pubs"
+    SET 
+        "updatedAt" = CURRENT_TIMESTAMP,
+        "title" = CASE 
+            WHEN TG_OP = 'DELETE' THEN NULL 
+            ELSE COALESCE(tmp."value", "pubs"."title")
+        END
+    FROM (SELECT DISTINCT "pubId" FROM inserted_deleted_rows) AS affected
+    LEFT JOIN tmp_affected_pubs tmp ON tmp."pubId" = affected."pubId"
+    WHERE "pubs"."id" = affected."pubId";
+
+    RETURN NULL;
```

Comment:
roughly:

for each `pubId` in the `inserted_deleted_rows`, set the `title` to the constructed `title` or the old one, and set the `updateAt` to the current timestamp.

note: this currently does not reflect changes from `value: "Some title"` -> `value: null`, as we don't really do that afaik. this is mostly bc this way we can use one update statement rather than 3ish.

is this a problem? should not be too hard to change it, i'll then probably just separate out updating the `updatedAt` and updating the `title`, add a separate query for setting the `DELETE` value, and then write a third query where change the `LEFT JOIN` to an inner join.
there might be a better way of doing this, just let me know whether changing a value to `null` should be accounted for!

Reply:
I think that's okay, but I'm not 100% sure. I think the only time we're using `value: null` is for relationship field values but since we haven't fully implemented those, it's hard to be certain that we'd never let that value change from something to `null`.